SU-SWS/stanford_profile
# D8CORE-687 Use SNOW API to set site name users

# READY FOR REVIEW

# Summary
- Created a plugin manager for easy to create install tasks
- Moved default content users to an install task
- Call SNOW API to get site provision data and set config_pages entity with appropriate values.
- https://github.com/SU-SWS/acsf-cardinalsites/pull/38

# Need Review By (Date)
- January

# Urgency
- low

# Steps to Test
1. `composer require su-sws/dev-D8CORE-687-snow-install`
1. in a `local.settings.php` add API username and password `putenv('STANFORD_SNOW_API_USER=****');` & `putenv('STANFORD_SNOW_API_PASS=****');'
1. fake being on ACSF:
   - edit the plugin file at `src/Plugin/InstallTask/SiteSettings.php` and in the method `isAhEnv()` return true.
1. run `blt drupal:install`
1. verify default users are still created.
1. verify site name has been set to "Default SNOW Site"
1. verify site email has been set to "sheamck@stanford.edu": `drush cget system.site mail --include-overridden`

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
